### DISCUSSION TITLE
Why username data is saved with quotes

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#12491] [Questions] [answered]
Hello,
I am fixing this problem that I ran into when reviewing the app and it is that after the signup operation, the metadata is sent by a trigger function to create a record in the `profiles` table. But when I check the `profiles` table I notice that username is saved with quotes.

the data type for username column is text

My trigger function is:

```
BEGIN
  INSERT INTO public.profiles(id, username, payku_account_user_id)
  VALUES (
    NEW.id,
    NEW.raw_user_meta_data -> 'username',
    NEW.raw_user_meta_data -> 'another_column_id'
  );
  RETURN NEW;
END;
```
In front the signup function is:

```
const createNewUser = async() => {
    const { username, email, password } = credentials;
    setLoading(true);
    try {
      const { user, error, session } = await supabase.auth.signUp({
          email: email,
          password: password,
        },
        {
          data: { 
            username: username, 
          }
        }
      );
      checkYourEmailToaster();
      
    } catch (err) {
      notifyError(err);
      setLoading(true);
    }
  }
```

**Comment by GaryAustin1** [comment, marked as answer]
You need to use `->>'username'` not `->` which returns a json object. `->>` returns text.
For the number it works either way.

**Reply by ricardosikic** [reply]
wow small difference!, thanks for the answer!
